php/libraries/Database.class.inc

```diff
@@ -777,8 +811,9 @@ class Database
     {
         $result = $this->pselectRow($query, $params);
         if (is_array($result) && count($result)) {
-
-            list(,$result) = each($result);
+            foreach ($result as $key => $value) {
+                $result = $value;
+            }
```

Comment:
This is failing PHPMD because the value '$key' is unused.

Also, the overloaded usage of the `$result` variable is pretty confusing. (I guess it always was, but for some reason it stands out more with the foreach loop.)

Wouldn't this be simpler (and more robust) if it just asserted `count($result) === 1` inside the if statement and returned right away?

Reply:
Will rewrite using array_values() instead of foreach(), this will remove the unneeded $key

As for count($result) === 1; if $result == false because the query returned no data, count($result) will return 1. Currently, it would pass the false already in $result up.